src/Categories/Bicategory/Monad/Bimodule.agda

```diff
@@ -0,0 +1,112 @@
+{-# OPTIONS --without-K --safe --lossy-unification #-}
+
+open import Categories.Bicategory
+
+module Categories.Bicategory.Monad.Bimodule {o ℓ e t} {𝒞 : Bicategory o ℓ e t} where
+
+open import Level
+open import Categories.Bicategory.Monad
+import Categories.Bicategory.Extras as Bicat
+open Bicat 𝒞
+
+record Bimodule (M₁ M₂ : Monad 𝒞) : Set (o ⊔ ℓ ⊔ e) where
+  open Monad M₁ renaming (C to C₁; T to T₁; μ to μ₁; η to η₁)
+  open Monad M₂ renaming (C to C₂; T to T₂; μ to μ₂; η to η₂)
+
+  field
+    F       : C₁ ⇒₁ C₂
+    actionˡ : F ∘₁ T₁ ⇒₂ F
+    actionʳ : T₂ ∘₁ F ⇒₂ F
+
+    assoc     : actionʳ ∘ᵥ (T₂ ▷ actionˡ) ∘ᵥ associator.from ≈ actionˡ ∘ᵥ (actionʳ ◁ T₁)
+    sym-assoc : actionˡ ∘ᵥ (actionʳ ◁ T₁) ∘ᵥ associator.to ≈ actionʳ ∘ᵥ (T₂ ▷ actionˡ)
+
+    assoc-actionˡ     : actionˡ ∘ᵥ (F ▷ μ₁) ∘ᵥ associator.from ≈ actionˡ ∘ᵥ (actionˡ ◁ T₁)
+    sym-assoc-actionˡ : actionˡ ∘ᵥ (actionˡ ◁ T₁) ∘ᵥ associator.to ≈ actionˡ ∘ᵥ (F ▷ μ₁)
+    assoc-actionʳ     : actionʳ ∘ᵥ (μ₂ ◁ F) ∘ᵥ associator.to ≈ actionʳ ∘ᵥ (T₂ ▷ actionʳ)
+    sym-assoc-actionʳ : actionʳ ∘ᵥ (T₂ ▷ actionʳ) ∘ᵥ associator.from ≈ actionʳ ∘ᵥ (μ₂ ◁ F)
+
+    identityˡ : actionˡ ∘ᵥ (F ▷ η₁) ∘ᵥ unitorʳ.to ≈ id₂
+    identityʳ : actionʳ ∘ᵥ (η₂ ◁ F) ∘ᵥ unitorˡ.to ≈ id₂
+
+id-bimodule : (M : Monad 𝒞) → Bimodule M M
+id-bimodule M = record
+  { F = T
+  ; actionˡ = μ
+  ; actionʳ = μ
+  ; assoc = assoc
+  ; sym-assoc = sym-assoc
+  ; assoc-actionˡ = assoc
+  ; sym-assoc-actionˡ = sym-assoc
+  ; assoc-actionʳ = sym-assoc
+  ; sym-assoc-actionʳ = assoc
+  ; identityˡ = identityˡ
+  ; identityʳ = identityʳ
+  }
+  where open Monad M
+
+record Bimodulehomomorphism {M₁ M₂ : Monad 𝒞} (B₁ B₂ : Bimodule M₁ M₂) : Set (ℓ ⊔ e) where
+  open Monad M₁ renaming (C to C₁; T to T₁)
+  open Monad M₂ renaming (C to C₂; T to T₂)
+  open Bimodule B₁ renaming (F to F₁; actionˡ to actionˡ₁; actionʳ to actionʳ₁)
+  open Bimodule B₂ renaming (F to F₂; actionˡ to actionˡ₂; actionʳ to actionʳ₂)
+  field
+    α       : F₁ ⇒₂ F₂
+    linearˡ : actionˡ₂ ∘ᵥ (α ◁ T₁) ≈ α ∘ᵥ actionˡ₁
+    linearʳ : actionʳ₂ ∘ᵥ (T₂ ▷ α) ≈ α ∘ᵥ actionʳ₁
+
+open import Categories.Category
+
+id-bimodule-hom : {M₁ M₂ : Monad 𝒞} → {B : Bimodule M₁ M₂} → Bimodulehomomorphism B B
+id-bimodule-hom {M₁} {M₂} {B} = record
```

Comment:
For type-checking speed and goal readability, it is a good idea to move the proofs (like the two `linear` ones below) to the `where` clause. Agda's propensity to eta-expand hurts when couple with constructions. [even more so for composition below.]